app/models/casebook/contents.rb

```diff
@@ -0,0 +1,102 @@
+# == Schema Information
+#
+# Table name: casebooks
+#
+#  id            :integer          not null, primary key
+#  title         :string           default("Untitled casebook"), not null
+#  slug          :string
+#  subtitle      :string
+#  headnote      :text
+#  public        :boolean          default(TRUE), not null
+#  root_id       :integer
+#  ordinals      :integer          default([]), not null, is an Array
+#  copy_of_id    :integer
+#  is_alias      :boolean
+#  material_type :string
+#  material_id   :integer
+#  created_at    :datetime         not null
+#  updated_at    :datetime         not null
+#
+
+class Casebook::Contents < Casebook::Generic
+  default_scope {where.not(root_id: nil)}
```

Comment:
The condition for a Book could be that `book_id = id`; it would make the same sense conceptually. There are a few reasons `null` is a more comfortable value for the root:

- Even though the primary key of a record shouldn't really change, so it's nbd, it's still not technically normalized to have the primary key duplicated in `book_id`.
- We want to be able to pull out a list of all Books; `book_id = null` is easier than `book_id = id`.
- We often want to be able to get a list of all contents of a book, not including the book; as null, we don't need the condition that `id != book_id`.

But on the flip side, finding the parent of a content node requires extra logic to check if the parent is a book. I don't feel super strongly about this, I guess I can see what it looks like that way.